tests/baselines/reference/asOperator1.js

```diff
@@ -0,0 +1,12 @@
+//// [asOperator1.ts]
+var as = 43;
+var x = undefined as number;
+var y = (null as string).length;
+var z = Date as any as string;
+
+
+//// [asOperator1.js]
+var as = 43;
+var x = undefined;
+var y = (null).length;

```

Comment:
We should do the same thing we do with parenthesized type assertions, and not emit the parens when we can.